sdk/communication/AzureCommunicationCalling/ApiSamples.md

```diff
@@ -0,0 +1,758 @@
+        options.videoOptions = VideoOptions(localVideoStream: self.localVideoStream!)!
+        let audioOptions = AudioOptions()!
+        audioOptions.muted = true // Join group call muted
+        options.audioOptions = audioOptions
+        if (threadId.starts(with: "http")) {
+            // Join a meeting with link
+            let teamsMeetingLinkLocator = TeamsMeetingLinkLocator(meetingLink: threadId)!;
+            self.call = self.callAgent?.join(with: teamsMeetingLinkLocator, joinCallOptions: options)
+        } else {
+            let groupCallLocator = GroupCallLocator(groupId: UUID(uuidString: threadId))!
+            self.call = self.callAgent?.join(with: groupCallLocator, joinCallOptions: options)
+        }
+        // Get updates on the call
+        self.call.delegate = self.callObserver
+    }
+}
+```
+
+* ### Mute/Unmute
+
+[Asynchronous] Local mute
+
+```swift
+self.call?.mute { (error) in
+    if error == nil {
+        print("Successfully muted")
+    } else {
+        print("Failed to mute")
+    }
+})
+
+```
+[Asynchronous] Local unmute
+
+```swift
+self.call?.unmute { (error) in
+    if error == nil {
+        print("Successfully unmuted")
+    } else {
+        print("Failed to unmute")
+    }
+})
+```
+
+* ### HangUp Call
+To hangUp the call:
+```swift
+var hangUpOptions = HangUpOptions()!
+hangUpOptions.forEveryOne = true
+self.call.hangUp(hangUpOptions) { (error) in
+    if (error == nil) {
+        print("HangUp successfull")
+    } else {
+        print("HangUp failed, try again")
+    }
+}
+print(CallObserver.callStateToString(state:call.state)) // => Disconnecting -> Disconnected
+```
+
+### Call Event Handling
+
+#### Event model
+Most of properties and collections can change it's value.
+To subscribe to these changes you can use following:
+* ##### Properties
+Application should implement the delegate to get notfied for change in properties
+```swift
+    self.call.delegate = CallObserver()
+    // Get the property of the call state by doing get on the call's state member
+    public func OnStateChanged(_ call: Call!,
+                                args: PropertyChangedEventArgs!)
+    {
+        print("Callback from SDK when the call state changes, current state: " + CallObserver.callStateToString(state:call.state))
+    }
+```
+
+* ##### Collections
+To subscribe to collection updated event also delegate needs to be implemented
+to get notfified when there is a change in collection
+```swift
+    self.call.delegate = self
+    // Collection contains the streams that were added or removed only
+    public func onLocalVideoStreamsChanged(_ call: Call!,
+                                           args: LocalVideoStreamsUpdatedEventArgs!)
+    {
+        print(args.addedStreams.count);
+        print(args.removedStreams.count);
+    }
+```
+
+To unsubscribe:
+```swift
+self.call?.delegate = nil // Will not recieve anymore call updates
```

Comment:
This feels very odd. Its not common practice to expect the developer to set the delegate to nil to unsubscribe to something

Reply:
Well this could be a valid scenario in some cases, say application does not want to receive any incoming calls. Then they can set the delegate to nil on CallAgent,  I understand that this is not a common thing application users do, we are just providing a way when you want to subscribe to an event, how to unsubscribe to an event, its completely up the application developer to use this or not.

Reply:
I don't think subscribe/unsubscribe is the right terminology for this pattern. Something like `NotificationCenter` would be a publish/subscribe type pattern. I think setting the `delegate` to nil here does make sense as part of the delegate pattern.

Reply:
What would be the suggested usage in this case ?
1. User of the application wants to get incoming call, so delegate is set and incoming call events is recvd.
2. Now when user of the application does not want to get incoming call notified, what would be the suggested approach that app developer needs to take ? If the CallAgent object is deleted and created and then not set the delegate it will work. But thats not a good solution , setting the delegate to nil then application will not get anymore incoming call notification.

Are you suggesting we should implement NotificationCenter support because we do not want app developers to set delegate to nil ?

Reply:
Will sync up with @JoshuaLai offline.